cmd/plume/prerelease.go

```diff
@@ -200,6 +278,45 @@ func getImageFile(client *http.Client, src *storage.Bucket, fileName string) (st
 	return imagePath, nil
 }
 
+func getFedoraImageFile(client *http.Client, src *storage.Bucket, fileName string) (string, error) {
+	cacheDir := filepath.Join(sdk.RepoCache(), "images", specChannel, specFedoraBoard, specFedoraVersion)
+	rawxzPath := filepath.Join(cacheDir, fileName)
+	imagePath := strings.TrimSuffix(rawxzPath, filepath.Ext(rawxzPath))
+
+	if _, err := os.Stat(imagePath); err == nil {
+		plog.Printf("Reusing existing image %q", imagePath)
+		return imagePath, nil
+	}
+
+	rawxzURI, err := url.Parse("https://kojipkgs.fedoraproject.org/compose/cloud/Fedora-Cloud-29-20181114.0/compose/Cloud/x86_64/images/Fedora-Cloud-Base-29-20181114.0.x86_64.raw.xz")
+	if err != nil {
+		return "", err
+	}
+
+	plog.Printf("Downloading image %q to %q", rawxzURI, rawxzPath)
+
+	if err := sdk.UpdateFile(rawxzPath, rawxzURI.String(), client); err != nil {
+		return "", err
+	}
+
+	//decompress it
+	plog.Printf("Decompressing %q...", rawxzPath)
+	if err := util.XZ2File(imagePath, rawxzPath); err != nil {
+		return "", err
+	}
+	return imagePath, nil
+}
+
+// getImageFile downloads a bzipped CoreOS image, verifies its signature,
+// decompresses it, and returns the decompressed path.
+func getImageFile(system string, client *http.Client, src *storage.Bucket, fileName string) (string, error) {
+	if system == "cl" {
```

Comment:
Minor nit: this is usual written as a `switch` statement with an explicit error default case.

Reply:
Done!